plugins/hotkeyshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,574 @@
+                return util.tableEquals(self.hotkeyshortcuts[hotkey], self.defaults[hotkey])
+            end,
+            callback = function()
+                self.hotkeyshortcuts[hotkey] = util.tableDeepCopy(self.defaults[hotkey])
+                self.updated = true
+            end,
+        })
+    end
+    table.insert(sub_items, {
+        text = _("No action"),
+        keep_menu_open = true,
+        separator = true,
+        checked_func = function()
+            return self.hotkeyshortcuts[hotkey] == nil
+        end,
+        callback = function()
+            self.hotkeyshortcuts[hotkey] = nil
+            self.updated = true
+        end,
+    })
+    Dispatcher:addSubMenu(self, sub_items, self.hotkeyshortcuts, hotkey)
+    -- Since we are already handling potential conflicts via overrideConflictingKeyEvents(), both "No action" and "Nothing",
+    -- introduced through Dispatcher:addSubMenu(), are effectively the same (from a user point of view); thus, we can do away
+    -- with "Nothing".
+    table.remove(sub_items, 3) -- removes the 'Nothing' option as it is redundant.
+    sub_items.max_per_page = 9 -- push settings ('Arrange actions', 'Show as quick menu', 'keep quick menu open') to page 2
+    return sub_items
+end
+
+function HotKeyShortcuts:genSubItem(hotkey, separator, hold_callback)
+    local reader_only = {
+        -- these button combinations are used by different events in FM already, don't allow users to customise them.
+        modifier_plus_down = true,
+        modifier_plus_left_page_back = true,
+        modifier_plus_left_page_forward = true,
+        modifier_plus_right_page_back = true,
+        modifier_plus_right_page_forward = true,
+        modifier_plus_press = true,
+    }
+    if self.settings_data.data["press_key_does_hotkeyshortcuts"] then
+        local do_not_allow_press_key_do_shortcuts_in_fm = { press = true }
+        util.tableMerge(reader_only, do_not_allow_press_key_do_shortcuts_in_fm)
+    end
+    local enabled_func
+    if reader_only[hotkey] then
+       enabled_func = function() return self.hotkey_mode == "hotkeyshortcuts_reader" end
+    end
+    return {
+        text_func = function() return self:shortcutTitleFunc(hotkey) end,
+        enabled_func = enabled_func,
+        sub_item_table_func = function() return self:genMenu(hotkey) end,
+        separator = separator,
+        hold_callback = hold_callback,
+        ignored_by_menu_search = true, -- This item is not strictly duplicated, but its subitems are. Ignoring it speeds up search.
+    }
+end
+
+function HotKeyShortcuts:genSubItemTable(hotkeyshortcuts)
+    local sub_item_table = {}
+    for _, item in ipairs(hotkeyshortcuts) do
+        table.insert(sub_item_table, self:genSubItem(item))
+    end
+    return sub_item_table
+end
+
+function HotKeyShortcuts:attachNewTableToExistingTable(orig_table, second_table)
+    for _, v in ipairs(second_table) do
+        table.insert(orig_table, v)
+    end
+end
```

Comment:
This could (should) be a local function (it does not use self, it's just a helper).
```lua
local function attachNewTableToExistingTable(orig_table, second_table)
...
end
```